HingeAngle/src/main/java/com/microsoft/device/display/samples/hingeangle/model/HingeAngleLiveData.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ *
+ */
+
+package com.microsoft.device.display.samples.hingeangle.model
+
+import android.app.Service
+import android.content.Context
+import android.hardware.SensorEvent
+import android.hardware.SensorManager
+import androidx.annotation.MainThread
+import androidx.lifecycle.LiveData
+import com.microsoft.device.display.samples.hingeangle.extensions.hingeAngleSensor
+import java.lang.IllegalStateException
+
+/**
+ * LiveData that exposes the hinge angle values from sensor
+ */
+class HingeAngleLiveData(context: Context) : LiveData<Int>() {
+    private val sensorManager = context.getSystemService(Service.SENSOR_SERVICE) as? SensorManager
+    private val hingeAngleSensor = sensorManager?.hingeAngleSensor
+
+    private val hingeAngleSensorListener = object : SensorEventListenerAdapter() {
```

Comment:
Can we maybe change this to a lambda?

Reply:
From what I know, only interfaces with one function can be converted to lambda. SensorEventListenerAdapter is a class adapter for SensorEventListener interface and this interface contains two functions (onSensorChanged and onAccuracyChanged). I added this adapter in order to have a clean code inside HingeAngleLiveData class, to avoid functions without implementations

Reply:
totally right! thanks for the detailed description!